az-digital/az_quickstart
# Closes #2572 Update Trellis Event Importer to use Trellis Event Search API

## Description
This PR updates the Trellis Event importer to use the new events search API provided by Trellis.

It provides this by means of a views bulk action form `AZEventTrellisViewsField extends BulkForm` that uses `views_remote_data` and an event subscriber to transparently fetch API results using `AZEventTrellisDataSubscriber` and add responses to the view results.

Updates to `TrellisHelper` have been made to allow for an API to search and fetch API responses.

A new view filter, `az_event_trellis_views_date_filter`, is provided that supports native Trellis date searching and translation of custom ranges into expected trellis values. (_Select or Other_ functionality)

Migration source is now provided by a new Source plugin `az_trellis_events_api`, which natively speaks to the Trellis API. This was necessary as the v2 Trellis API requires POST requests with a JSON body, and the native migrate url tools are primarily based around GET.

- [x] Initial skeleton of import structure with bulk form
- [x] Migration data fetcher
- [x] Implement all filters
- [x] Helper API additions
- [x] Migration updates

## Related issues
Closes #2572 

## How to test

- enable `az_event_trellis`
- enable `az_enterprise_attributes_import` (optionally)
- visit `/admin/trellis-event-importer`
- select filter options, click `search`
- verify that results are correct
- choose events to import by clicking the checkbox to the left of the event row
- click `save` at the bottom of the form
- verify that the event is imported

**Note: the current QA endpoint is not hooked up to the live data. Events that exist on production data are not necessarily expected to be in the results**

## Types of changes


### Arizona Quickstart (install profile, custom modules, custom theme)
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal core
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

### Drupal contrib projects
- **Patch release changes**
- **Minor release changes**
- **Other or unknown**

## Checklist
